### PR TITLE
Expose `TreeDataProvider`s in API

### PR DESCRIPTION
**Pull request by alexweininger**
* Exposing the workspace and application tree data providers so they can be used by client extensions.
* Some cleanup and renaming around the registration functions

### CODE REVIEW

File: src/api/v2/v2AzureResourcesApi.ts [hunk 1]
```diff
@@ -179,6 +179,10 @@ export interface ResourcePickOptions {
  * The current (v2) Azure Resources extension API.
  */
 export interface V2AzureResourcesApi extends AzureResourcesApiBase {
+
+    applicationResourceTreeDataProvider: vscode.TreeDataProvider<unknown>;
```
Comment (bwateratmsft) [inline src/api/v2/v2AzureResourcesApi.ts:183]: For the purposes of the API, these can probably be `vscode.TreeDataProvider<ResourceGroupsItem>` or something along those lines.

File: src/api/v2/v2AzureResourcesApiWrapper.ts [hunk 2]
```diff
@@ -17,6 +17,14 @@ export class V2AzureResourcesApiWrapper implements V2AzureResourcesApi {
         return this.api.apiVersion;
     }
 
+    get applicationResourceTreeDataProvider(): vscode.TreeDataProvider<unknown> {
+        return this.callWithTelemetryAndErrorHandlingSync('v2.getApplicationResourceTreeDataProvider', () => this.api.applicationResourceTreeDataProvider);
```
Comment (bwateratmsft) [inline src/api/v2/v2AzureResourcesApiWrapper.ts:21]: Does a simple `get` method really deserve telemetry wrapping?

File: src/api/v2/v2AzureResourcesApi.ts (hunk 1 above)
Comment (alexweininger) [reply src/api/v2/v2AzureResourcesApi.ts:183]: Ok. I wasn't sure how much we wanted to hide from the clients.

File: src/api/v2/v2AzureResourcesApiWrapper.ts (hunk 2 above)
Comment (alexweininger) [reply src/api/v2/v2AzureResourcesApiWrapper.ts:21]: Ya... it's probably overkill

File: src/api/v2/v2AzureResourcesApi.ts (hunk 1 above)
Comment (bwateratmsft) [reply src/api/v2/v2AzureResourcesApi.ts:183]: I think they'll almost _need_ at least something, e.g. to indicate that the objects are wrappers. What did we say as the input to the quick pick experience? Isn't that `ResourceGroupsItem`?

Comment (alexweininger) [reply src/api/v2/v2AzureResourcesApi.ts:183]: Currently the quick pick experience needs a `TreeDataProvider<unknown>`

**Pull request merged**